lib/components/narrative/line-itin/connected-itinerary-body.js

```diff
@@ -60,14 +61,43 @@ class ConnectedItineraryBody extends Component {
       timeOptions
     } = this.props
     const { LegIcon } = this.context
+    const clonedItinerary = clone(itinerary)
+
+    // Support OTP1 flex messages in Trip Details
+    // Adding empty pickupBookingInfo and dropOffBookingInfo objects
+    // to a leg will trigger relevant flex pickup / dropoff descriptions in
+    // the Trip Details component.
+    const boardingRuleLeg = clonedItinerary.legs.find(
+      (leg) => !!leg.boardRule && !leg.pickupBookingInfo
+    )
+    const alightingRuleLeg = clonedItinerary.legs.find(
+      (leg) => !!leg.alightRule && !leg.dropOffBookingInfo
+    )
+
+    // Add pickupBookingInfo for any boarding rules
+    if (boardingRuleLeg) {
+      boardingRuleLeg.pickupBookingInfo = {
```

Comment:
Is this adjusting the `clonedItinerary`? I think this is what's causing this to no longer work

Reply:
Things still seem to be working for me? Where are you seeing things not working? 

here's a sample search when running nysdot-qa that works for me: [here](http://localhost:9966/#/?ui_activeSearch=5yrwwpa68&ui_activeItinerary=0&fromPlace=Fitbody%20by%20Krista%2C%20979%20RT-22%2C%20Brewster%2C%20NY%2C%2010509%2C%20USA%3A%3A41.39683320215821%2C-73.60353447614618&toPlace=Terry%20Hill%20Rd%2C%20Carmel%2C%20NY%2C%2010512%2C%20USA%3A%3A41.461234603414816%2C-73.6545398793056&date=2022-03-28&time=13%3A38&arriveBy=false&mode=WALK%2CBUS%2CRAIL%2CSUBWAY%2CTRAM%2CFERRY&showIntermediateStops=true&maxWalkDistance=1207&optimize=QUICK&walkSpeed=1.34&ignoreRealtimeUpdates=true&numItineraries=3&otherThanPreferredRoutesPenalty=900)